src/Provider/Clock.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace League\OAuth2\Client\Provider;
+
+/**
+ * Represents an implementation of a Clock.
+ */
+class Clock
+{
+
+  /**
+   * Get the current time.
+   *
+   * @return \DateTimeImmutable
+   */
+    public function now()
```

Comment:
Maybe this method should be `getTime()` instead of `now()`, so that Clock could return any DateTime object that an implementer wants to set (fixed, now, etc.)?

Reply:
I used `now` as thats the same method implemented by the reference clock projects linked in summary. Also a `getTime` is strictly speaking not appropriate as you're getting more than just **time**.

Reply:
`now()` is my preference.